File: common/src/main/java/com/ultreon/mods/advanceddebug/client/menu/pages/DefaultPage.java
package com.ultreon.mods.advanceddebug.client.menu.pages;

import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage;
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext;
import com.ultreon.mods.advanceddebug.client.Config;
import com.ultreon.mods.advanceddebug.mixin.common.MinecraftAccessor;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.client.server.IntegratedServer;
import org.jetbrains.annotations.NotNull;

public class DefaultPage extends DebugPage {
    private final Minecraft mc = Minecraft.getInstance();

    public DefaultPage() {
        super();
    }

    @Override
    public void render(@NotNull GuiGraphics gfx, IDebugRenderContext ctx) {
        if (Config.SHOW_FPS_ON_DEFAULT_PAGE.get()) {
            ctx.left("FPS", MinecraftAccessor.getFps());
            IntegratedServer server;
            if (mc.hasSingleplayerServer() && (server = mc.getSingleplayerServer()) != null) {
                ctx.left("Server TPS", server.getTickCount());
            }
        }
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/client/menu/pages/DefaultPage.kt
package com.ultreon.mods.advanceddebug.client.menu.pages

import com.mojang.blaze3d.vertex.PoseStack
import com.ultreon.mods.advanceddebug.AdvancedDebug
import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext
import com.ultreon.mods.advanceddebug.client.Config
import com.ultreon.mods.advanceddebug.mixin.common.MinecraftAccessor

class DefaultPage : DebugPage(AdvancedDebug.modId, "default") {
    override fun render(poseStack: PoseStack, ctx: IDebugRenderContext) {
        if (Config.showFpsOnDefaultPage.get()) {
            ctx.left("FPS", MinecraftAccessor.getFps())
            if (minecraft.hasSingleplayerServer()) {
                minecraft.singleplayerServer?.let { server ->
                    ctx.left("Server TPS", server.tickCount)
                }
            }
        }
    }
}